stubs/openpyxl/openpyxl/packaging/relationship.pyi

```diff
@@ -37,5 +44,20 @@ class RelationshipList(Serialisable):
     def to_tree(self): ...
 
 def get_rels_path(path): ...
-def get_dependents(archive, filename): ...
-def get_rel(archive, deps, id: Incomplete | None = None, cls: Incomplete | None = None): ...
+def get_dependents(archive: ZipFile, filename: str) -> RelationshipList: ...
+
+# If `id` is None, `cls` needs to have ClassVar `rel_type`.
+# The `deps` attribute used at runtime is for internal use immediatly after the return.
+# `cls` cannot be None
+@overload
+def get_rel(
+    archive: ZipFile, deps: RelationshipList, id: None = None, *, cls: type[_SerialisableRelTypeT]
+) -> _SerialisableRelTypeT | None: ...
+@overload
+def get_rel(
+    archive: ZipFile, deps: RelationshipList, id: None, cls: type[_SerialisableRelTypeT]
+) -> _SerialisableRelTypeT | None: ...
+@overload
+def get_rel(archive: ZipFile, deps: RelationshipList, id: str, *, cls: type[_SerialisableT]) -> _SerialisableT: ...
```

Comment:
Isn't this redundant with the next overload?

Reply:
I wanna say you're right? Probably a leftover from before I realized `cls` wasn't optional.

Reply:
I wanna say you're right? Probably a leftover from before I realized `cls` wasn't optional.

Reply:
Not sure how doable it is, but I suggested catching this at mypy: https://github.com/python/mypy/issues/16164